Proposition for OverlayLocation

DalwynWasTaken opened the pull request:
Instead of using points, using a custom location class to handle positioning for the Overlay.
This would be the property of the Overlay that the ConfigManager would store.

**This is a proposition so feel free to just say no. If it gets approved, I'll slowly work through all the files to make everything is done and that OverlayPosition is a useful class**

DalwynWasTaken committed:
Proposition for OverlayLocation

Instead of using points, using a custom location class to handle positioning for the Overlay.

This would be the property of the Overlay that the ConfigManager would store.

magicus commented:
It is probably a good idea to have a custom class to define overlay positions. I think overlay positions should be in terms of "snapping" to different sides, and sizes/position might be given in percentages of the screen size, rather than pixels. That way, it will be more useful when switching between computers/window sizes.

DalwynWasTaken commented:
Width and height (sizes) should be kept in pixels from my experience with drawing elements. It can have adverse effects if that is stored in a percentage.

DalwynWasTaken committed:
Touch-Ups

Made some fields public to allow for cleaner code in certain areas.

Renamed x and y to allow for cleaner calling and it being implied.

Added comments to remove confusion if this is ever looked back at.
